router/image/templates/nginx.conf

```diff
@@ -141,47 +146,44 @@ http {
         {{ end }}
     }
     ## end deis-store-gateway
-
+    {{ $useSSL := "false" }}
+    {{ if exists "/deis/router/sslCert" }}{{ $useSSL := getv "/deis/router/sslCert" }}{{ end }}
+    {{ $domains := ls "/deis/domains" }}
+    {{ $certs := ls "/deis/certs" }}
     ## start service definitions for each application
-    {{ $affinityArg := .deis_router_affinityArg }}
-    {{ $certs := .deis_certs }}
-    {{ $domains := .deis_domains }}
-    {{ $root := . }}
-    {{ range $service := .deis_services }}{{ if $service.Nodes }}
-    upstream {{ Base $service.Key }} {
-        {{ if $affinityArg }}hash $arg_{{ $affinityArg }} consistent;
+    {{ range $app := lsdir "/deis/services" }}
+    {{ $upstreams := printf "/deis/services/%s/*" $app}}
+    upstream {{ $app }} {
+        {{ if exists "/deis/router/affinityArg" }}
+        hash $arg_{{ getv "/deis/router/affinityArg" }} consistent;
         {{ end }}
-        {{ range $upstream := $service.Nodes }}server {{ $upstream.Value }};
+        {{ range gets $upstreams }}server {{ .Value }};
         {{ end }}
     }
-    {{ end }}
-
+    {{ $appContainers := gets $upstreams }}{{ $appContainerLen := len $appContainers }}
     ## server entries for custom domains
-    {{ range $domain := $domains }}{{ if eq (Base $service.Key) $domain.Value }}
+    {{ range $app_domain := $domains }}{{ if eq $app (getv (printf "/deis/domains/%s" $app_domain)) }}
     server {
-        server_name {{ Base $domain.Key }};
+        server_name {{ $app_domain }};
         {{/* if a SSL certificate is installed for this domain, use SSL */}}
-        {{/* NOTE (bacongobbler): domains are separate from the default platform domain, */}}
+        {{/* Note (bacongobbler): domains are separate from the default platform domain, */}}
         {{/* so we can't rely on deis.conf as each domain is an island */}}
-        {{/* FIXME (bacongobbler): confd turns hyphens to dashes, so we need to account */}}
-        {{/* for that in domains */}}
-        {{ if index $root (printf "deis_certs_%s_cert" (Replace (Base $domain.Key) "-" "_" -1)) }}
+        {{ range $cert := $certs }}{{ if eq $app_domain $cert }}
```

Comment:
this behaviour was reversed with the last rebase (from #3519). Would you mind fixing that up please?